### PR TITLE
[Feature Fix] Render Special Characters Onboarder

### LINKED ISSUE #2958: Onboarder renders some special characters in HTML
<h4>Steps</h4>
1. Using the onboarder on the user dashboard, search for a project in the "Go to my project" field or in the "search for a project" field in the file upload section of the onboarder—be sure this project has an ampersand (&) in its name.

<h4>Outcome</h4>
1. Some, but not all project titles render the ampersand in HTML. Percent signs, exclamation points, and dollar signs all render properly.

screen shot 2015-06-04 at 10 13 43 am

screen shot 2015-06-04 at 10 14 01 am

<h4>Expected Outcome</h4>
1. Special characters should not be rendered in HTML anywhere on the OSF.

### PR DESCRIPTION
**Pull request by haoyuchen1992**
## Purpose
The html-encoded titles aren't decoded when they are shown on board. 
screen shot 2015-06-04 at 12 18 19 pm
Resolved #2958 

## Change
1) Add html decode function in osfHelper.js
2) Apply html decode function to the title shown on board.
screen shot 2015-06-04 at 2 18 53 pm

## Side Effect
None. Some people warn that JQuery HTML decoder will add the risk of inserting Javascript or HTML code in this title [link](http://stackoverflow.com/questions/5796718/html-entity-decode). But in our Project Creator API-V1, it has already delete the Javascript or HTML tag. So when our tile will never have executable html and javascript tags. The risk is not true in our case.

### CODE REVIEW

**Comment by caileyfitz** [comment]
@haoyuchen1992 will this also fix the same problem when searching for a project in the Add Links modal on a project overview page? 
screen shot 2015-06-04 at 3 02 57 pm

**Comment by caileyfitz** [comment]
Ah, just looked at the code and it looks like this is just applied to the onboarder.

**Comment by samanehsan** [comment]
Yes, I think this just applies to the onboarder.

**Comment by haoyuchen1992** [comment]
@caileyfitz  In our database, currently all the titles are stored as html-encoded string. So maybe there are  a lot of other places which get the encoded string and display without decoding.

**Comment by lbanner** [comment]
Quite possible -- Let's leave this PR focused as is, and we can add additional issues as we assess. Thanks.

**Comment by samanehsan** [comment]
Code looks good :+1: @sloria, I think this is good to go.

File: website/static/js/osfHelpers.js [hunk 1]
```diff
@@ -417,6 +417,14 @@ var htmlEscape = function(text) {
     return $('<div/>').text(text).html();
 };
 
+
+/**
+ * Decode Escaped html characters in a string.
+ */
+var htmlDecode = function(text) {
```
Comment (sloria) [inline website/static/js/osfHelpers.js]: @haoyuchen1992 Please add unit tests for this function in `osfHelpers.test.js`.

**Comment by sloria** [comment]
@haoyuchen1992 Pass finished. Once the tests are added, this is good to merge. 

QA: Make sure to test this with "unsafe" inputs--script injection, etc. @caileyfitz

File: website/static/js/osfHelpers.js (hunk 1 above)
Comment (haoyuchen1992) [inline website/static/js/osfHelpers.js]: @sloria Just add unit tests.

Comment (sloria) [inline website/static/js/osfHelpers.js]: Thanks for the quick response. This is good to merge once the CI build passes.

**Pull request merged**